src/darwin/Framework/CHIP/CHIPDeviceController.mm

```diff
@@ -338,52 +337,17 @@ - (void)_serviceEvents
             return;
         }
 
-        __block fd_set readFDs, writeFDs, exceptFDs;
-        struct timeval aSleepTime;
-        int numFDs = 0;
-        aSleepTime.tv_sec = 5;
-
-        FD_ZERO(&readFDs);
-        FD_ZERO(&writeFDs);
-        FD_ZERO(&exceptFDs);
-
-        chip::System::Layer * systemLayer = NULL;
-        chip::Inet::InetLayer * inetLayer = NULL;
-        // ask for the system and inet layers
-
         [self.lock lock];
-        self.cppController->GetLayers(&systemLayer, &inetLayer);
-
-        if (systemLayer != NULL && systemLayer->State() == chip::System::kLayerState_Initialized) {
-            systemLayer->PrepareSelect(numFDs, &readFDs, &writeFDs, &exceptFDs, aSleepTime);
-        }
-
-        if (inetLayer != NULL && inetLayer->State == chip::Inet::InetLayer::kState_Initialized) {
-            inetLayer->PrepareSelect(numFDs, &readFDs, &writeFDs, &exceptFDs, aSleepTime);
-        }
 
-        [self.lock unlock];
-
-        int selectRes = select(numFDs, &readFDs, &writeFDs, &exceptFDs, const_cast<struct timeval *>(&aSleepTime));
-
-        [self.lock lock];
         if (!self.cppController->IsConnected()) {
             [self.lock unlock];
             // cancel the loop, it'll restart the next time a connection is established
             return;
         }
-        systemLayer = NULL;
-        inetLayer = NULL;
-        self.cppController->GetLayers(&systemLayer, &inetLayer);
 
-        if (systemLayer != NULL && systemLayer->State() == chip::System::kLayerState_Initialized) {
-            systemLayer->HandleSelectResult(selectRes, &readFDs, &writeFDs, &exceptFDs);
-        }
-
-        if (inetLayer != NULL && inetLayer->State == chip::Inet::InetLayer::kState_Initialized) {
-            inetLayer->HandleSelectResult(selectRes, &readFDs, &writeFDs, &exceptFDs);
-        }
+        self.cppController->ServiceEvents();
```

Comment:
@vivien-apple this was structured in the way it was for a specific reason. We don't want to hold the lock for the entire `select` duration. 

So the code was structured to grab the chip locks for a short time before going and waiting inside the selectQueue.

Reply:
Ah right. I guess I will have to add it back with some #ifdef and a comment then. Sorry for the noise.

Reply:
Why the #ifdef, sorry I might be missing some context too. :)

Reply:
I am trying to enable the NW.Framework backend in the demo app.
But PrepareSelect/HandleSelectResult are guarded by CHIP_SYSTEM_CONFIG_USE_SOCKETS

I'm working on a patch to fix that on the platform side (since I need parts of ServiceEvents since system timers like the secure session expiry relies on it if I'm correct) but in the meantime I was hoping to have a version of the iOS demo app that I can use as if, compatible with the sockets backend and the NW backend.

Reply:
Okay yeah i suspected as much. Makes sense! 😄